src/wallet-connect.ts

```diff
@@ -1,36 +1,69 @@
+import { createWeb3Modal } from '@web3modal/wagmi/react';
+import { EIP6963Connector } from '@web3modal/wagmi';
 import { configureChains, createConfig } from 'wagmi';
-import { EthereumClient, w3mConnectors } from '@web3modal/ethereum';
-import { jsonRpcProvider } from '@wagmi/core/providers/jsonRpc';
 import { mainnet, hardhat } from 'wagmi/chains';
+import { jsonRpcProvider } from '@wagmi/core/providers/jsonRpc';
+
+import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet';
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
 
 if (!process.env.REACT_APP_PROJECT_ID) {
   throw new Error('Missing REACT_APP_PROJECT_ID env variable');
 }
 
+if (!process.env.REACT_APP_MAINNET_RPC_URL) {
+  throw new Error('Missing REACT_APP_MAINNET_RPC_URL env variable');
+}
+
 export const projectId = process.env.REACT_APP_PROJECT_ID;
 
-const chains = [mainnet, hardhat];
-
-const { publicClient } = configureChains(chains, [
-  // In the web3modal docs they use their "w3mProvider", which prefers using their RPC proxy for a set number of chains,
-  // and falls back to using a "jsonRpcProvider" (like below). It is unclear what the use of the RPC proxy is, and it seems
-  // like a central point of failure, so we rather go with a "jsonRpcProvider" and use the RPC providers directly.
-  // See: https://docs.walletconnect.com/2.0/web3modal/react/installation
-  // See: https://github.com/WalletConnect/web3modal/blob/V2/chains/ethereum/src/utils.ts
-  jsonRpcProvider({
-    rpc: (chain) => {
-      return {
-        http: chain.rpcUrls.default.http[0]!,
-        webSocket: chain.rpcUrls.default.webSocket?.[0],
-      };
-    },
-  }),
-]);
-
-export const wagmiClient = createConfig({
+const chainInfos = [
+  { chain: mainnet, rpcUrl: process.env.REACT_APP_MAINNET_RPC_URL },
```

Comment:
The advantage is that the RPC url is managed completely by us. There is still potential for this RPC URL to be down (limit exceeded or the provider having issues), but we can react more easily to this, e.g. by re-deploying the app with a different RPC.